From 0c284ee7945c9db73aa307d33be5ac855073ed46 Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Mon, 11 Oct 2021 14:09:54 +0200
Subject: [PATCH] window: Make the Today button adaptive

This makes the Today button use an icon when the window is narrow,
allowing it to reach narrower widths.
---
 src/gui/gcal-window.ui | 46 ++++++++++++++++++++++++++++++++++--------
 1 file changed, 38 insertions(+), 8 deletions(-)

diff --git a/src/gui/gcal-window.ui b/src/gui/gcal-window.ui
index 3434ce54..13e94976 100644
--- a/src/gui/gcal-window.ui
+++ b/src/gui/gcal-window.ui
@@ -226,15 +226,45 @@
               </object>
             </child>
             <child>
-              <object class="GtkButton" id="today_button">
-                <property name="label" translatable="yes">Today</property>
+              <object class="HdySqueezer" id="today_button">
                 <property name="visible">True</property>
-                <property name="can_focus">True</property>
-                <property name="receives_default">True</property>
-                <property name="action-name">win.today</property>
-                <accelerator key="t" modifiers="GDK_CONTROL_MASK" signal="clicked"/>
-                <accelerator key="Down" modifiers="GDK_MOD1_MASK" signal="clicked"/>
-                <accelerator key="Home" signal="clicked"/>
+                <property name="transition-type">crossfade</property>
+                <property name="xalign">0</property>
+                <child>
+                  <object class="GtkButton">
+                    <property name="label" translatable="yes">Today</property>
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="receives_default">True</property>
+                    <property name="action-name">win.today</property>
+                    <property name="sensitive" bind-source="today_button" bind-property="sensitive" bind-flags="sync-create"/>
+                    <accelerator key="t" modifiers="GDK_CONTROL_MASK" signal="clicked"/>
+                    <accelerator key="Down" modifiers="GDK_MOD1_MASK" signal="clicked"/>
+                    <accelerator key="Home" signal="clicked"/>
+                  </object>
+                </child>
+                <child>
+                  <object class="GtkButton">
+                    <property name="visible">True</property>
+                    <property name="can_focus">True</property>
+                    <property name="receives_default">True</property>
+                    <property name="action-name">win.today</property>
+                    <property name="sensitive" bind-source="today_button" bind-property="sensitive" bind-flags="sync-create"/>
+                    <property name="halign">start</property>
+                    <accelerator key="t" modifiers="GDK_CONTROL_MASK" signal="clicked"/>
+                    <accelerator key="Down" modifiers="GDK_MOD1_MASK" signal="clicked"/>
+                    <accelerator key="Home" signal="clicked"/>
+                    <style>
+                      <class name="image-button"/>
+                    </style>
+                    <child>
+                      <object class="GtkImage">
+                        <property name="visible">True</property>
+                        <property name="icon_name">calendar-today-symbolic</property>
+                      </object>
+                    </child>
+                  </object>
+                </child>
               </object>
             </child>
             <child type="title">
-- 
2.30.2

